### ISSUE TITLE
Spliting channels during loss calculation and combine them individually

### ISSUE THREAD

**Issue by Harris91-lee** [open-mmlab/mmsegmentation#3555]
I'm using the mmseg for 2D segmentation for medical data and im getting a good results using dice loss. I have mainly three labels : 0,1, and 2, background, label1, and label2. In the decode_head.py we have the segmentation loss calculation part: 

        for loss_decode in losses_decode:
            if loss_decode.loss_name not in loss:
                loss[loss_decode.loss_name] = loss_decode(
                    seg_logits,
                    seg_label,
                    weight=seg_weight,
                    ignore_index=self.ignore_index)
            else:
                loss[loss_decode.loss_name] += loss_decode(
                    seg_logits,
                    seg_label,
                    weight=seg_weight,
                    ignore_index=self.ignore_index)
          loss['acc_seg'] = accuracy(
          seg_logits, seg_label, ignore_index=self.ignore_index)
          return loss

originally in my dataset, seg_logits.shape [B,C,W,H] is [4,3,512,512], and  seg_label.shape is [4,1,512,512], **however**, now im trying to split the  seg_logits to be seg_logits1, seg_logits2, and seg_logits3 for each channel so im getting [4,1,512,512], and respectively i split the seg_label by its value, i.e, seg_label1, seg_label2, seg_label3 and shape [4,1,512,512] for each. 
but how can calculate the losses of each separately? I tried this  but im getting strange values: 
     
      for loss_decode in losses_decode:
           if loss_decode.loss_name not in loss2:
               loss2[loss_decode.loss_name] = loss_decode(
                   seg_logits2,
                   seg_labels2,
                   weight=seg_weight2,
                   ignore_index=self.ignore_index)
           else:
               loss2[loss_decode.loss_name] += loss_decode(
                   seg_logits2,
                   seg_labels2,
                   weight=seg_weight2,
                   ignore_index=self.ignore_index)

        for loss_decode in losses_decode:
           if loss_decode.loss_name not in loss1:
               loss1[loss_decode.loss_name] = loss_decode(
                   seg_logits1,
                   seg_labels1,
                   weight=seg_weight1,
                   ignore_index=self.ignore_index)
           else:
               loss1[loss_decode.loss_name] += loss_decode(
                   seg_logits1,
                   seg_labels1,
                   weight=seg_weight1,
                   ignore_index=self.ignore_index)

           loss = loss1[loss_decode.loss_name] + loss2[loss_decode.loss_name] 
the value of the loss is 0 or very small like 0.1 so no matter how long i train the value doesnt change, can you guys help me understanding where im doin mistake here? Thank you in advnave!
